src/exodus/test/sp_tests.cpp

```diff
@@ -0,0 +1,102 @@
+#include "../sp.h"
+
+#include "test/test_bitcoin.h"
+#include "main.h"
+#include "miner.h"
+#include "wallet/wallet.h"
+
+#include <boost/test/unit_test.hpp>
+
+struct SpTestingSetup : public TestingSetup
+{
+    SpTestingSetup()
+        : TestingSetup(CBaseChainParams::REGTEST), spinfo(pathTemp / "MP_spinfo_test", false)
+    {
+    }
+
+    int GenerateEmptyBlock(size_t blocks)
```

Comment:
If we follow my idea we don't need this method here.

Reply:
still need this but simplify it  by calling exists function.